File: chroma-league/src/test/java/com/bonepl/chromaleague/hud/animations/RiftAnimationTest.java
package com.bonepl.chromaleague.hud.animations;

import com.bonepl.chromaleague.GameStateMocks;
import com.bonepl.chromaleague.hud.AnimationTester;
import com.bonepl.chromaleague.rest.gamestats.GameStats;
import com.bonepl.chromaleague.rest.gamestats.MapTerrain;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.EnumSource;

import static org.mockito.Mockito.when;

class RiftAnimationTest {

    @ParameterizedTest
    @EnumSource(MapTerrain.class)
    void playRiftAnimation(MapTerrain mapTerrain) {
        // given
        GameStats gameStats = new GameStateMocks().gameStats();
        when(gameStats.mapTerrain()).thenReturn(mapTerrain.getApiType());

        // then
        new AnimationTester().testAnimation(new RiftAnimation());
    }
}

File: test/hud/animations/RiftAnimation.test.ts
import {describe, test} from 'vitest';
import {AnimationTester} from '../../AnimationTester.js';
import {GameStateMocks} from '../../GameStateMocks.js';
import {RiftAnimation} from '../../../src/hud/animations/RiftAnimation.js';

const MAP_TERRAINS = ['Chemtech', 'Cloud', 'Hextech', 'Infernal', 'Mountain', 'Ocean'] as const;

/**
 * Integration test for rift terrain animations on actual hardware.
 */
describe('RiftAnimation Integration', () => {
    test.each(MAP_TERRAINS)('playRiftAnimation - %s', async (mapTerrain) => {
        // given
        const gameStats = new GameStateMocks().gameStats();
        gameStats.mapTerrain = mapTerrain;

        // then
        await new AnimationTester().testAnimation(new RiftAnimation());
    }, 30000);
});
